src/libs/antares/study/parts/hydro/hydromaxtimeseriesreader.h

```diff
@@ -0,0 +1,83 @@
+/*
+** Copyright 2007-2023 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+
+#ifndef __ANTARES_LIBS_STUDY_PARTS_HYDRO_MAX_TIME_SERIES_READER_H__
+#define __ANTARES_LIBS_STUDY_PARTS_HYDRO_MAX_TIME_SERIES_READER_H__
+
+#include "../../../array/antares//array/matrix.h"
+
+namespace Antares::Data
+{
+/*!
+**  This class provides support for old studies, reading from deprecated files,
+**  fils matrix dailyMaxPumpAndGen and transfers data to the corresponding data
+**  class members of class PartHydro. Just versions below 8.7 will use instance
+**  of this class to be compatible with current implementation.
+*/
+class HydroMaxTimeSeriesReader
```

Comment:
1.**maxDailyGenEnergy** and **maxDailyPumpEnergy** are not part of **DataSeriesHydro** class because we can not have multiple TS's like for hydro storage or ROR.
2.We can make them private.
3.We have old file with 4x365 matrix and we are loading that data in the **dailyMaxPumpAndGen** matrix. From **dailyMaxPumpAndGen** matrix we are distributing data to the other 4 destinations (every column of the **dailyMaxPumpAndGen** matrix will be copied to the one of the 4 destinations). That destinations are two matrices from the **PartHydro** class **maxDailyGenEnergy** and **maxDailyPumpEnergy** and other two matrices from the **DataSeriesHydro** class **maxHourlyGenPower** and **maxHourlyPumpPower**.  These four matrices (destinations) are used in solver, and they must be filled with appropriate data and if it is a case of an old study those data are coming from **dailyMaxPumpAndGen**  matrix, which we load with deprecated file (maxpower_areaid.txt). Don't see a way without coping these data, because we don't want to use  **dailyMaxPumpAndGen** directly in solver, because that would complicate code even more.
4.Regarding other two points, that can be implemented also.

Reply:
Addition to the third statement. If we had load function for class **Matrix** that can receive as argument range of the columns that we want to load, not to load whole txt file, just columns that we want, into our **Matrix** instance, than this **HydroReader** class would be useless, because than we could load data from the deprecated file directly to our 4 destinations **(maxDailyGenEnergy, maxDailyPumpEnergy, maxHourlyGenPower, maxHourlyPumpPower)**, and we wouldn't need **dailyMaxPumpAndGen** matrix to distribute data.

Reply:
- Answer to **1.**  : we don't have to do it now, but it would seem that the natural place for **maxDailyGenEnergy** and **maxDailyPumpEnergy** is **DataSeriesHydro** (at least, I can't see a reason why not).
- Answer to **3.** : you're right, I missed that point.
- Answer to the very last comment : **absolutely**, but unfortunately, it seems impossible. As you said, the only real mission of this class is to load a 4 columns matrix and extract the columns we're interested in. Apart from that, we wouldn't need a class.

Reply:
My idea is to remove copy functions from HydroReader class (copy function are used to copy data to the **maxDailyGenEnergy** and **maxDailyPumpEnergy**) and to place them in the PartHydro class, like you did for the **maxHourlyGenPower** and **maxHourlyPumpPower** because of the **feature envy**. Then we will have something like this:
![image](https://github.com/AntaresSimulatorTeam/Antares_Simulator/assets/131546833/036911fa-2051-43c7-ab91-61588277ec6b)

Reply:
Honestly, I don't know yet what to think about it. 
The problem seems more global : we should think at the **hydro time series** scale.
I mean that hydro part has several time series, and several ways to load them (depending on which is the study version, if the loader is UI or Solver, and more [related to TS generation]). 
What should we do to first clarify this first.
Imagining a more global way to load would help, I think.

And there is the **how** we do it. Do we want external TS loader (and saver) ? This means that we would empty class   **PartHydro** from all loading and saving functions. This could be a way to respect more the **single responsibility principle**, which requires small classes. And **PartHydro** has many lines of code.

I'm sorry I can't more specific. 

About the change you suggest, the loading part would stay externalized, but the copy would be internalized (done in **PartHydro**), right ? What would be the benefit of this ?

Reply:
```cpp
loadForVersionXXX(); // old
loadForVersionYYY(); // new
convertFromXXXtoYYY();

load(int version) {
  if(version <= XXX)
  {
    loadForVersionXXX();
    convertFromXXXtoYYY();
  }
  else
    loadForVersionYYY();
}
```
To be discussed